File: nextjs_space/audit_all_images.ts
import { PrismaClient } from '@prisma/client';
import { config } from 'dotenv';
import fs from 'fs';
import path from 'path';

config();

const prisma = new PrismaClient();

async function main() {
  console.log('🔍 COMPREHENSIVE IMAGE AUDIT\n');
  console.log('='.repeat(80));

  // Load products_data.json
  const jsonPath = path.join(__dirname, 'data', 'products_data.json');
  const productsData = JSON.parse(fs.readFileSync(jsonPath, 'utf-8'));

  // Get products from database
  const dbProducts = await prisma.product.findMany({
    where: { active: true },
    select: {
      sku: true,
      title_en: true,
      supplier: true,
      images: true
    }
  });

  const issues: Array<{
    sku: string;
    title: string;
    supplier: string;
    issue: string;
    jsonImage: string;
    dbImage: string;
  }> = [];

  const summary: Record<string, {
    total: number;
    official: number;
    cdn: number;
    local: number;
  }> = {};

  console.log('\n📊 CHECKING ALL PRODUCTS:\n');

  dbProducts.forEach(dbProduct => {
    const jsonProduct = productsData.find((p: any) => p.sku === dbProduct.sku);
    const supplier = dbProduct.supplier || 'Unknown';
    
    if (!summary[supplier]) {
      summary[supplier] = { total: 0, official: 0, cdn: 0, local: 0 };
    }
    summary[supplier].total++;

    const dbImages = dbProduct.images as string[];
    const jsonImages = jsonProduct?.images || [];

    const dbImage = dbImages[0] || '';
    const jsonImage = jsonImages[0] || '';

    // Categorize image source
    if (dbImage.startsWith('/')) {
      summary[supplier].local++;
    } else if (dbImage.includes('abacusaicdn') || dbImage.includes('cdn.abacus.ai')) {
      summary[supplier].cdn++;
      
      // Check if JSON has official URL but DB has CDN
      if (jsonImage && !jsonImage.includes('abacus') && jsonImage.startsWith('http')) {
        issues.push({
          sku: dbProduct.sku,
          title: dbProduct.title_en,
          supplier,
          issue: 'DB has CDN but JSON has official URL',
          jsonImage,
          dbImage
        });
      }
    } else if (dbImage.startsWith('http')) {
      summary[supplier].official++;
    }

    // Check Placid Instruments specifically
    if (supplier === 'Placid Instruments') {
      if (!jsonImage.includes('placidinstruments.com') && !dbImage.startsWith('/')) {
        if (jsonImage.includes('abacus')) {
          issues.push({
            sku: dbProduct.sku,
            title: dbProduct.title_en,
            supplier,
            issue: 'Using CDN instead of placidinstruments.com',
            jsonImage,
            dbImage
          });
        }
      }
    }
  });

  console.log('\n📈 SUMMARY BY SUPPLIER:\n');
  Object.entries(summary)
    .sort((a, b) => b[1].total - a[1].total)
    .forEach(([supplier, stats]) => {
      console.log(`${supplier}:`);
      console.log(`  Total: ${stats.total}`);
      console.log(`  ✓ Official: ${stats.official}`);
      console.log(`  📦 Local: ${stats.local}`);
      console.log(`  🌐 CDN: ${stats.cdn}`);
      console.log('');
    });

  if (issues.length > 0) {
    console.log('\n⚠️  ISSUES FOUND:\n');
    console.log(`Total issues: ${issues.length}\n`);

    // Group by supplier
    const groupedIssues: Record<string, typeof issues> = {};
    issues.forEach(issue => {
      if (!groupedIssues[issue.supplier]) {
        groupedIssues[issue.supplier] = [];
      }
      groupedIssues[issue.supplier].push(issue);
    });

    Object.entries(groupedIssues).forEach(([supplier, supplierIssues]) => {
      console.log(`\n${supplier} (${supplierIssues.length} issues):`);
      console.log('-'.repeat(80));
      supplierIssues.forEach((issue, i) => {
        console.log(`\n${i + 1}. ${issue.sku} - ${issue.title}`);
        console.log(`   Issue: ${issue.issue}`);
        console.log(`   JSON:  ${issue.jsonImage}`);
        console.log(`   DB:    ${issue.dbImage}`);
      });
    });

    // Save to file
    const reportPath = path.join(__dirname, '..', '..', 'Uploads', 'IMAGE-AUDIT-DETAILED.json');
    fs.writeFileSync(reportPath, JSON.stringify(issues, null, 2));
    console.log(`\n\n📄 Detailed report saved to: ${reportPath}`);
  } else {
    console.log('\n✅ NO ISSUES FOUND! All images are properly sourced.\n');
  }
}

main()
  .catch(console.error)
  .finally(() => prisma.$disconnect());

File: nextjs_space/data/audit_all_images.py
import json
import re
from collections import defaultdict

with open('products_data.json', 'r', encoding='utf-8') as f:
    products = json.load(f)

# Group products by supplier
suppliers = defaultdict(list)
for p in products:
    supplier = p.get('supplier', 'Unknown')
    suppliers[supplier].append(p)

print("\n" + "="*80)
print("IMAGE AUDIT REPORT - ALL PRODUCTS")
print("="*80)

# Check for problematic images
issues = []
for supplier, prods in sorted(suppliers.items()):
    print(f"\n\n### {supplier} ({len(prods)} products) ###\n")
    
    for p in prods[:5]:  # Show first 5 from each supplier
        sku = p.get('sku', 'NO_SKU')
        title = p.get('title_en', 'NO_TITLE')[:50]
        images = p.get('images', [])
        
        if not images:
            issues.append({'sku': sku, 'supplier': supplier, 'issue': 'NO_IMAGES'})
            print(f"❌ {sku:30} | {title:50} | NO IMAGES")
        else:
            img = images[0]
            # Check if image is a local file or external URL
            if img.startswith('/'):
                print(f"✓  {sku:30} | {title:50} | Local: {img}")
            elif 'http' in img:
                # Extract domain
                domain_match = re.search(r'https?://([^/]+)', img)
                domain = domain_match.group(1) if domain_match else 'unknown'
                print(f"🌐 {sku:30} | {title:50} | External: {domain}")
            else:
                issues.append({'sku': sku, 'supplier': supplier, 'issue': 'INVALID_IMAGE_PATH'})
                print(f"❌ {sku:30} | {title:50} | Invalid: {img}")

print(f"\n\n{'='*80}")
print(f"ISSUES FOUND: {len(issues)}")
print("="*80)
for issue in issues[:20]:
    print(f"{issue['sku']:30} | {issue['supplier']:30} | {issue['issue']}")

# Summary by supplier
print(f"\n\n{'='*80}")
print("SUMMARY BY SUPPLIER")
print("="*80)
for supplier, prods in sorted(suppliers.items()):
    with_images = sum(1 for p in prods if p.get('images'))
    local_images = sum(1 for p in prods if p.get('images') and p['images'][0].startswith('/'))
    external_images = sum(1 for p in prods if p.get('images') and 'http' in p['images'][0])
    
    print(f"{supplier:30} | Total: {len(prods):3} | With Images: {with_images:3} | Local: {local_images:3} | External: {external_images:3}")

